src/Stack/Package.hs

```diff
@@ -693,7 +693,7 @@ packageDescModulesAndFiles pkg = do
             (mapM
                  (asModuleAndFileMap benchComponent benchmarkFiles)
                  (benchmarks pkg))
-    dfiles <- resolveGlobFiles
+    dfiles <- resolveGlobFiles (specVersion pkg)
                     (extraSrcFiles pkg
                         ++ map (dataDir pkg FilePath.</>) (dataFiles pkg))
```

Comment:
While I'm looking at this code I just want to point out that this will misfire if `dataDir` has an asterisk in it anywhere. That's not new in this patch. `*` isn't a legitimate path character on Windows anyway, so it's mostly a theoretical concern.